IOMC/ParticleGuns/src/CloseByParticleGunProducer.cc

```diff
@@ -81,7 +82,12 @@ void CloseByParticleGunProducer::produce(Event& e, const EventSetup& es) {
     } else
       phi += fDelta / fR;
 
-    double fEn = CLHEP::RandFlat::shoot(engine, fEnMin, fEnMax);
+    double fEn;
+    if (numParticles > 1 && fMaxEnSpread)
+      fEn = fEnMin + ip * (fEnMax - fEnMin) / (numParticles - 1);
+    else
+      fEn = CLHEP::RandFlat::shoot(engine, fEnMin, fEnMax);
+
```

Comment:
can you explain what has been done here?

Reply:
@SiewYan, in case the newly introduced flag (`MaxEnSpread`) is true, the energies of the generated particles are set in linear steps of the specified range `(fEnMin, fEnMax)`; otherwise they are computed randomly in the specified range, as in the original code.

Reply:
thanks, its all look good to me